colossalai/inference/core/engine.py

```diff
@@ -348,6 +357,10 @@ def enable_spec_dec(
         engine.clear_spec_dec()
         ```
         """
+        self.logger.warning(
+            f"Current method will be deprecated soon. To use speculative decoding, please set `use_spec_dec` in `InferenceConfig` instead."
+        )
+
```

Comment:
At this moment, there exists only one method of running speculative decoding: `enable_spec_dec`. A deprecated warning is not expected here as this is the only way to turn on the feature and further development (e.g. API) has not been kicked off.

Reply:
Ok. Removed.